yewprint-doc/src/icon/mod.rs

```diff
@@ -2,39 +2,187 @@ mod example;
 
 use crate::ExampleContainer;
 use example::*;
+use once_cell::sync::Lazy;
+use std::borrow::Cow;
+use web_sys::HtmlInputElement;
 use yew::prelude::*;
-use yewprint::H1;
+use yewprint::{HtmlSelect, Icon, IconName, InputGroup, Intent, Slider, Text, H1, H5};
 
-pub struct IconDoc;
+pub struct IconDoc {
+    callback: Callback<ExampleProps>,
+    state: ExampleProps,
+    search_string: String,
+}
+
+#[derive(Clone)]
+pub enum IconDocMsg {
+    Example(ExampleProps),
+    SearchIcon(String),
+}
+
+static ICON_LIST: Lazy<Vec<(String, IconName)>> = Lazy::new(|| {
+    IconName::ALL
+        .iter()
+        .map(|x| (format!("{:?}", x).to_lowercase(), *x))
+        .collect()
+});
+
+fn get_icons_from_name(name: &str) -> impl Iterator<Item = IconName> {
+    let name = name.to_lowercase();
+    ICON_LIST
+        .iter()
+        .filter_map(move |(icon_name, icon)| icon_name.contains(&name).then_some(*icon))
+}
 
 impl Component for IconDoc {
-    type Message = ();
+    type Message = IconDocMsg;
     type Properties = ();
 
-    fn create(_ctx: &Context<Self>) -> Self {
-        IconDoc
+    fn create(ctx: &Context<Self>) -> Self {
+        IconDoc {
+            callback: ctx.link().callback(|x| IconDocMsg::Example(x)),
+            state: ExampleProps {
+                icon_name: IconName::Print,
+                intent: None,
+                icon_size: 16,
+            },
+            search_string: Default::default(),
+        }
     }
 
-    fn update(&mut self, _ctx: &Context<Self>, _msg: Self::Message) -> bool {
+    fn update(&mut self, _ctx: &Context<Self>, msg: Self::Message) -> bool {
+        match msg {
+            IconDocMsg::Example(x) => self.state = x,
+            IconDocMsg::SearchIcon(x) => self.search_string = x,
+        }
         true
     }
 
-    fn view(&self, _ctx: &Context<Self>) -> Html {
+    fn view(&self, ctx: &Context<Self>) -> Html {
+        let example_props = self.state.clone();
         let source = crate::include_raw_html!(
             concat!(env!("OUT_DIR"), "/", file!(), ".html"),
             "bp3-code-block"
         );
 
+        let icon_list = get_icons_from_name(&self.search_string.to_lowercase())
+            .map(|x| {
```

Comment:
let's see what happens if you get rid of this useless function
```suggestion
        let search_string = self.search_string.to_lowercase();
        let icon_list = ICON_LIST
            .iter()
            .filter_map(|(icon_name, icon)| icon_name.contains(&search_string).then_some(*icon))
            .map(|x| {
```